### PR TITLE
Properly cleanup event listeners on the Internal Provider.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
We had a small memory leak that became especially apparent when the app was disconnected and attempting to reconnect and call `eth_chainID` over long periods of time.

This issue _might_ be related to https://github.com/tallycash/extension/issues/1158

### CODE REVIEW

**Review by mhluongo — APPROVED** [review]
Looks good.

File: background/redux-slices/utils/contract-utils.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const internalProviderPort = {
     this.listeners.push(listener)
   },
   removeEventListener(toRemove: (message: any) => unknown): void {
-    this.listeners.filter((listener) => listener !== toRemove)
+    this.listeners = this.listeners.filter((listener) => listener !== toRemove)
```
Comment (Shadowfiend) [inline background/redux-slices/utils/contract-utils.ts:22]: Oof.

Comment (mhluongo) [reply background/redux-slices/utils/contract-utils.ts:22]: What am I missing here? This is an intentional state mutation...?

Comment (mhluongo) [reply background/redux-slices/utils/contract-utils.ts:22]: Are we looking for a `splice()` or something else that will preserve the reference? And if so, why?

**Pull request merged**